src/compiler/scala/tools/nsc/typechecker/Typers.scala

```diff
@@ -3333,15 +3333,14 @@ trait Typers extends Adaptations with Tags with TypersTracking with PatternTyper
           def handleOverloaded = {
             val undetparams = context.undetparams
 
-            def funArgTypes(tps: List[Type]) = tps.map { tp =>
-              val relTp = tp.asSeenFrom(pre, fun.symbol.owner)
+            def funArgTypes(relTps: List[Type]) = relTps.map { relTp =>
               val argTps = functionOrSamArgTypes(relTp)
               //println(s"funArgTypes $argTps from $relTp")
               argTps.map(approximateAbstracts)
             }
 
-            def functionProto(argTps: List[Type]): Type =
-              try functionType(funArgTypes(argTps).transpose.map(lub), WildcardType)
+            def functionProto(argTpWithAlt: List[(Type, Symbol)]): Type =
+              try functionType(funArgTypes(argTpWithAlt.map { case (tp, alt) => tp.asSeenFrom(pre, alt.owner) }).transpose.map(lub), WildcardType)
```

Comment:
It's such a detail that I'm sorry to go back and forth so many times.. I just don't understand.. IIUC adriaan suggests to pass `argTpWithAlt` as argument to `funArgTypes` directly and do the `asSeenFrom` there, which also looks better to me. You even seem to agree, as you say

  > Moving it out [of `funArgTypes`?] didn't really make the code any simpler

so why did you move it out?